kernel/mallocator.cpp

```
@@ -0,0 +1,79 @@
+#include "mallocator.hpp"
+
+#include <bolgenos-ng/stdtypes.hpp>
+
+#include <bolgenos-ng/error.h>
+
+
+
+#include <bolgenos-ng/cout.hpp>
+
+namespace {
+
+size_t chain_unit_size(size_t index) {
+	if (index == 0)
+		return 8;
+	return index * 16;
+}
+
+} // namespace
+
+void memory::allocators::Mallocator::initialize(
+		memory::allocators::PageAllocator *fallback,
+		size_t chain_memory, size_t chain_length) {
+	fallback_ = fallback;
+	chain_length_ = chain_length;
+	const size_t chain_size = sizeof(SlabAllocator)*chain_length;
+	if (! internal_allocator_.initialize(chain_size, 1)) {
```

:exclamation: The function should return an error insead of panicing.